internal/provider/credentials.go

```diff
@@ -19,11 +19,11 @@ const (
 	endPointSTSDefault = "https://sts.amazonaws.com"
 )
 
-func configureCredentialsSTS(cfg config.Config, secret, region, awsAccessKeyID, awsSecretAccessKey, awsSessionToken, endpoint string) (config.Config, error) {
+func configureCredentialsSTS(cfg *config.Config, secret, region, awsAccessKeyID, awsSecretAccessKey, awsSessionToken, endpoint string) (config.Config, error) {
```

Comment:
curious if this change is needed for the scope of this PR or it's an opportunistic improvement